Src/Particle/AMReX_ParticleTransformation.H

```diff
@@ -394,12 +394,55 @@ auto filterParticles (DstTile& dst, const SrcTile& src, Pred&& p,
     amrex::ParallelForRNG(np,
     [=] AMREX_GPU_DEVICE (int i, amrex::RandomEngine const& engine) noexcept
     {
-        p_mask[i] = particle_detail::call_f(p, src_data, i, engine);
+        p_mask[i] = particle_detail::call_f(p, src_data, src_start+i, engine);
     });
-
+    Gpu::streamSynchronize();
     return filterParticles(dst, src, mask.dataPtr(), src_start, dst_start, n);
 }
 
+/**
+ * \brief Conditionally copy particles from src to dst based on the value of mask.
+ * A transformation will also be applied to the particles on copy.
+ *
+ * \tparam DstTile the dst particle tile type
+ * \tparam SrcTile the src particle tile type
+ * \tparam Index the index type, e.g. unsigned int
+ * \tparam F the transform function type
+ *
+ * \param dst the destination tile
+ * \param src the source tile
+ * \param mask pointer to the mask - 1 means copy, 0 means don't copy
+ * \param f defines the transformation that will be applied to the particles on copy
+ *
+ */
+template <typename DstTile, typename SrcTile, typename Index, typename F,
+          std::enable_if_t<std::is_integral<Index>::value, int> foo = 0>
+Index filterAndTransformParticles (DstTile& dst, const SrcTile& src, Index* mask, F&& f,
+                                   Index src_start, Index dst_start) noexcept
```

Comment:
Could the already existing `filterAndTransformParticles` function, that doesn't take `src_start` and `dst_start`, now be implemented in terms of this function (and likewise for the other overload you added below)?

Reply:
Pushed update commit that uses the overloaded function